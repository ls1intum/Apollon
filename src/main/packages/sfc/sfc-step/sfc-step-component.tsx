import { ThemedRect } from '../../../components/theme/themedComponents';
import { Text } from '../../../components/controls/text/text';
import React from 'react';
import { SfcEnd } from '../sfc-end/sfc-end';

export function SfcStepComponent({ element }: { element: SfcEnd }) {
  return (
    <g>
      <ThemedRect fillColor={element.fillColor} x="0" y="0" width="100%" height="100%" />
      <Text x="50%" y="50%" fill={element.textColor} textAnchor="middle" dominantBaseline="middle">
        {element.name}
      </Text>
      <ThemedRect x="0" y="0" fillColor="none" width="100%" height="100%" strokeColor={element.strokeColor} />
    </g>
  );
}
