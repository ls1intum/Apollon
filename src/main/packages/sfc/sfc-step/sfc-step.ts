import { SfcContainer } from '../base/sfc-container';
import { UMLElementType } from '../../uml-element-type';

/**
 * Represents a step element in a sfc.
 */
export class SfcStep extends SfcContainer {
  type = UMLElementType.SfcStep;
}
