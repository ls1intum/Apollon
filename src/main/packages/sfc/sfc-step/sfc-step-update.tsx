import React, { ComponentClass } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import styled from 'styled-components';
import { ColorButton } from '../../../components/controls/color-button/color-button';
import { Divider } from '../../../components/controls/divider/divider';
import { Textfield } from '../../../components/controls/textfield/textfield';
import { I18nContext } from '../../../components/i18n/i18n-context';
import { localized } from '../../../components/i18n/localized';
import { ModelState } from '../../../components/store/model-state';
import { StylePane } from '../../../components/style-pane/style-pane';
import { UMLElementRepository } from '../../../services/uml-element/uml-element-repository';
import { SfcStep } from './sfc-step';

const Flex = styled.div`
  display: flex;
  align-items: baseline;
  justify-content: space-between;
`;

interface OwnProps {
  element: SfcStep;
}

type StateProps = {};

interface DispatchProps {
  update: typeof UMLElementRepository.update;
  delete: typeof UMLElementRepository.delete;
}

type Props = OwnProps & StateProps & DispatchProps & I18nContext;

function SfcStepUpdateComponent({ element, update }: Props) {
  const [colorOpen, setColorOpen] = React.useState(false);

  const toggleColor = () => {
    setColorOpen(!colorOpen);
  };

  const rename = (name: string) => {
    update(element.id, { name });
  };

  return (
    <div>
      <section>
        <Flex>
          <Textfield value={element.name} onChange={rename} autoFocus />
          <ColorButton onClick={toggleColor} />
        </Flex>
        <StylePane open={colorOpen} element={element} onColorChange={update} fillColor lineColor textColor />
        <Divider />
      </section>
    </div>
  );
}

const enhance = compose<ComponentClass<OwnProps>>(
  localized,
  connect<{}, DispatchProps, OwnProps, ModelState>(null, {
    update: UMLElementRepository.update,
    delete: UMLElementRepository.delete,
  }),
);

export const SfcStepUpdate = enhance(SfcStepUpdateComponent);
