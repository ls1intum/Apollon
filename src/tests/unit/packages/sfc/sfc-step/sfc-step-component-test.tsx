import { wrappedRender } from '../../../test-utils/render';
import * as React from 'react';
import { SfcStep } from '../../../../../main/packages/sfc/sfc-step/sfc-step';
import { SfcStepComponent } from '../../../../../main/packages/sfc/sfc-step/sfc-step-component';

it('render the sfc-step-component', () => {
  const step: SfcStep = new SfcStep({ name: 'Step' });
  const { getByText, baseElement } = wrappedRender(
    <svg>
      <SfcStepComponent element={step} />
    </svg>,
  );
  expect(getByText(step.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
