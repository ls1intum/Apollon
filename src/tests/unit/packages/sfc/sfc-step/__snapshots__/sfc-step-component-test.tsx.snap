// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the sfc-step-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <rect
          class="sc-fqkvVR lnbwZS"
          fill="white"
          height="100%"
          stroke="none"
          width="100%"
          x="0"
          y="0"
        />
        <text
          dominant-baseline="middle"
          font-weight="bold"
          pointer-events="none"
          text-anchor="middle"
          x="50%"
          y="50%"
        >
          Step
        </text>
        <rect
          class="sc-fqkvVR jRtNsY"
          fill="none"
          height="100%"
          stroke="black"
          width="100%"
          x="0"
          y="0"
        />
      </g>
    </svg>
  </div>
</body>
`;
